lib/jose-utils.js

```diff
@@ -219,6 +219,75 @@ Utils.stringFromArray = function(arr) {
 };
 
 /**
+ * Converts string into array of UTF-8 bytes. Implementation from Google closure:
+ * https://github.com/google/closure-library
+ * @param str string
+ * @return Uint8Array
+ */
+Utils.utf8BytesFromString = function(str) {
+  var out = [], p = 0;
+  for (var i = 0; i < str.length; i++) {
+    var c = str.charCodeAt(i);
+    if (c < 128) {
+      out[p++] = c;
+    } else if (c < 2048) {
+      out[p++] = (c >> 6) | 192;
+      out[p++] = (c & 63) | 128;
+    } else if (
+      ((c & 0xFC00) == 0xD800) && (i + 1) < str.length &&
+      ((str.charCodeAt(i + 1) & 0xFC00) == 0xDC00)) {
+      // Surrogate Pair
+      c = 0x10000 + ((c & 0x03FF) << 10) + (str.charCodeAt(++i) & 0x03FF);
+      out[p++] = (c >> 18) | 240;
+      out[p++] = ((c >> 12) & 63) | 128;
+      out[p++] = ((c >> 6) & 63) | 128;
+      out[p++] = (c & 63) | 128;
+    } else {
+      out[p++] = (c >> 12) | 224;
+      out[p++] = ((c >> 6) & 63) | 128;
+      out[p++] = (c & 63) | 128;
+    }
+  }
+  return new Uint8Array(out);
+};
+
+/**
+ * Converts UTF-8 byte array into string. Implementation from Google closure:
+ * https://github.com/google/closure-library
+ * @param utf8 arrayish
+ * @return string
+ */
+Utils.stringFromUtf8Bytes = function(utf8) {
```

Comment:
Same comment here. Convert every element into "%" + hex + "%" + hex + ... and call decodeURIComponent() on the resulting string.